knotx-core/knotx-template-engine/src/test/java/com/cognifide/knotx/engine/ServiceCorrectConfigurationTest.java

```diff
@@ -17,47 +17,63 @@
  */
 package com.cognifide.knotx.engine;
 
+import com.cognifide.knotx.engine.TemplateEngineConfiguration.ServiceMetadata;
+
 import org.hamcrest.CoreMatchers;
 import org.junit.Before;
 import org.junit.Test;
 
+import java.util.stream.Collectors;
+
+import io.vertx.core.json.JsonArray;
+
 import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertEquals;
 
 public class ServiceCorrectConfigurationTest extends AbstractKnotxConfigurationTest {
 
-  private String CORRECT_JSON = "service-correct.json";
+  private static final String CORRECT_JSON = "service-correct.json";
 
   private TemplateEngineConfiguration correctConfig;
 
-  private TemplateEngineConfiguration.ServiceMetadata expectedServiceOne;
-  private TemplateEngineConfiguration.ServiceMetadata expectedServiceTwo;
+  private ServiceMetadata expectedServiceOne;
+  private ServiceMetadata expectedServiceTwo;
 
   @Before
   public void setUp() throws Exception {
     correctConfig = new TemplateEngineConfiguration(readConfig(CORRECT_JSON));
-
-    expectedServiceOne = createMockedService("/service/mock/.*", "localhost", 3000);
-    expectedServiceTwo = createMockedService("/service/.*", "localhost", 8080);
+    expectedServiceOne = createMockedService("/service/mock/.*", "localhost", 3000, new JsonArray().add("Accept-*").add("Location"));
+    expectedServiceTwo = createMockedService("/service/.*", "localhost", 8080, new JsonArray());
   }
 
   @Test
-  public void testConfigBeanInitializedProperly() {
+  public void whenCorrectConfigIsProvided_expectConfigIsProperlyParsed() {
```

Comment:
Missing few very important tests here:

- [ ] verify - wildcard "*" is properly working (e.g. X*, X-Some-*, etc.)
- [ ] verify - no headers filtered out if "*"
- [ ] verify  - all headers filtered out if nothing is defined (just empty "allowed.headers" : {} )
- [ ] verify -  all headers filtered out if no "allowed.headers" in config